policykit/policyengine/views.py

```diff
@@ -824,80 +824,567 @@ def document_action_recover(request):
 
     return HttpResponse()
 
-def policy_from_request (request, key_name = 'policy'):
+def policy_from_request(request, key_name = 'policy'):
     policy_id = request.GET.get(key_name)
-
     from policyengine.models import Policy
-
-    # Get source policy object
     try:
         return Policy.objects.get(pk=policy_id)
     except Policy.DoesNotExist:
         raise Http404("Policy does not exist")
 
-def embed_initial(request):
-    # Get source policy based on id passed via the URL
-    policy_source = policy_from_request(request, key_name="source")
 
-    # Variables without a default value or value are set in the first step of the flow
-    initial_variables = policy_source.variables.filter(default_value__exact="", value__exact="")
+#===
```

Comment:
ahh yeah I don't like the fact that we have app-specific code in the policyengine folder or that we have integration specific stuff here too (with Slack). For the purpose of the deployment tmrw I'll let it slide but we should change this ASAP.